root/edit/details/EditWork.js

```diff
@@ -0,0 +1,157 @@
+/*
+ * @flow
+ * Copyright (C) 2020 Anirudh Jain
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import DescriptiveLink from
+  '../../static/scripts/common/components/DescriptiveLink';
+import WordDiff from '../../static/scripts/edit/components/edit/WordDiff';
+import {INSERT, DELETE} from '../../static/scripts/edit/utility/editDiff';
+import Diff from '../../static/scripts/edit/components/edit/Diff';
+import DiffSide from '../../static/scripts/edit/components/edit/DiffSide';
+import FullChangeDiff from
+  '../../static/scripts/edit/components/edit/FullChangeDiff';
+import {commaOnlyListText} from
+  '../../static/scripts/common/i18n/commaOnlyList';
+
+type Props = {
+  ...EditT,
+  +data: {
+    new: {
+      attributes?: {
+        [attributeName: string]: $ReadOnlyArray<WorkAttributeT>,
+      },
+      languages: $ReadOnlyArray<string>,
+    },
+  },
+  +display_data: {
+    attributes?: {
+      [attributeName: string]: CompT<$ReadOnlyArray<string>>,
+    },
+    comment?: CompT<string | null>,
+    iswc?: CompT<string | null>,
+    language?: CompT<LanguageT | null>,
+    languages?: CompT<$ReadOnlyArray<string>>,
+    name?: CompT<string>,
+    type?: CompT<WorkTypeT | null>,
+    work: WorkT,
+  },
+};
+
+const EditWork = ({edit}: {edit: Props}) => {
+  const display = edit.display_data;
+  const comment = display.comment;
+  const iswc = display.iswc;
+  const name = display.name;
+  const type = display.type;
+  const language = display.language;
+  const languages = display.languages;
+  const attributes = display.attributes ?? {};
+  return (
+    <table className="details edit-work">
+      <tr>
+        <th>{addColonText(l('Work'))}</th>
+        <td colSpan="2">
+          <DescriptiveLink entity={display.work} />
+        </td>
+      </tr>
+      {name ? (
+        <WordDiff
+          label={addColonText(l('Name'))}
+          newText={name.new}
+          oldText={name.old}
+        />
+      ) : null}
+      {comment ? (
+        <WordDiff
+          label={addColonText(l('Disambiguation'))}
+          newText={comment?.new ?? ''}
+          oldText={comment?.old ?? ''}
+        />
+      ) : null}
+      {iswc ? (
+        <Diff
+          label={addColonText(l('ISWC'))}
+          newText={iswc?.new ?? ''}
+          oldText={iswc?.old ?? ''}
+        />
+      ) : null}
+      {type ? (
+        <FullChangeDiff
+          label={addColonText(l('Work type'))}
+          newContent={lp_attributes(type?.new?.name ?? '', 'work_type')}
+          oldContent={lp_attributes(type?.old?.name ?? '', 'work_type')}
```

Comment:
You already know type exists, so you probably don't need to do type? here, right? Same for iswc above, etc.

Also, the way you've done this here crashes the edit display (as usual, please do test the stuff locally on the site!). You need to do the same you did on https://github.com/metabrainz/musicbrainz-server/pull/1362/files#diff-fc0b1391e46fee0ae9a544b658540ee3R68

Reply:
Sure I did test all the stuff locally. I'll recheck the same. :)